### ISSUE TITLE
Is it possible merge http://www.regexper.com/

### ISSUE THREAD

**Issue by ghost** [gskinner/regexr#52]
The feature regexper.com is very good as well.
Would you please merge it into regexr?

The graphic explain about regex code. is very clean and clear

**Comment by stuartpb** [comment]
I could see this being fairly useful as an integrated feature into the top line, actually, as it does give another way of reviewing the expression that is arguably neater than the mouseover hints RegExr currently provides.

**Comment by penne12** [comment]
+1

**Comment by gskinner** [comment]
How does this compare to our existing Explain feature. What does it offer you that Explain doesn't? Trying to establish a value proposition for this before committing time to any implementation.

**Comment by stuartpb** [comment]
As I said [two and a half years ago](https://github.com/gskinner/regexr/issues/52#issuecomment-69702237), it gives another way of reviewing the expression that's neater than the mouseover explanations: Regexper's diagrams allow the structure of the *entire expression* to be read *at once*, in a way that is *directly annotated* (for instance, the way "One of:" is placed above a box denoting a character class, instead of having this notion be something implicit that could slip through the cracks in a developer's conception of their regexp when reviewing its parts, piece by piece, in a tooltip).

**Comment by gskinner** [comment]
Right, and following that input we added the Explain tool, which isn't exactly the same, but does provide the single view overview you suggested.

My updated question was in relation to Explain. Does it do what you need? Or are there still aspects of regexper that you miss?

**Comment by gskinner** [comment]
https://jex.im/regulex/ is another, nearly identical tool to look at.

**Comment by stuartpb** [comment]
Ah, okay, I see. Yeah, the "Explain" overview is still too wordy, too sprawling, and not adequately visual (compare the looping arrow around "word" on https://regexper.com/#(%5BA-Z%5D)%5Cw%2B against the abstract cognitive load Explain uses to present the concept: "Plus. Match 1 or more of the preceding token.").

**Comment by stuartpb** [comment]
And, yes, the Regulex codebase looks much cleaner to integrate (it's also MIT licensed); that'd be the way I'd integrate this feature today.

**Comment by stuartpb** [comment]
One thing I'll say is that, from a UX standpoint, it'd make a lot more sense to me to have this Diagram and/or Explain tool up at the top, next to "Save" and "Share", than at the bottom, beneath the Text that it's not relevant to at all.
